src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -58,7 +58,7 @@
 
 ansi_dialect.set_lexer_matchers(
     [
-        RegexLexer("whitespace", r"[\t ]+", WhitespaceSegment),
+        RegexLexer("whitespace", r"[^\S\r\n]+", WhitespaceSegment),
```

Comment:
Could we also add a comment here to the effect of:
```
# Match all forms of whitespace except newlines and carriage returns:
# https://stackoverflow.com/questions/3469080/match-whitespace-but-not-newlines
# This pattern allows us to also match non-breaking spaces (#2189).
```
so people don't question our choice. 😄  

Also the whitespace pattern gets used in a couple different places so we should update these too:
https://github.com/sqlfluff/sqlfluff/blob/a5378224e2f1d0441137645179dfeb076dccc642/src/sqlfluff/dialects/dialect_ansi.py#L79

https://github.com/sqlfluff/sqlfluff/blob/a5378224e2f1d0441137645179dfeb076dccc642/src/sqlfluff/dialects/dialect_tsql.py#L123

https://github.com/sqlfluff/sqlfluff/blob/a5378224e2f1d0441137645179dfeb076dccc642/test/core/parser/lexer_test.py#L86

https://github.com/sqlfluff/sqlfluff/blob/a5378224e2f1d0441137645179dfeb076dccc642/test/core/parser/lexer_test.py#L88

Reply:
@jpy-git  got them!